### PR TITLE
[SPARK-42476][CONNECT][DOCS] Complete Spark Connect API reference

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to complete missing API Reference for Spark Connect.

Built API docs should include "Changed in version" for Spark Connect when it's implemented as below:

Screen Shot 2023-02-20 at 9 49 09 AM

### Why are the changes needed?

Improving usability for Spark Connect.

### Does this PR introduce _any_ user-facing change?

No, it's documentation.

### How was this patch tested?

Manually built docs, confirmed each function and class one by one.

### CODE REVIEW

File: python/docs/source/reference/pyspark.connect/functions.rst [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    specific language governing permissions and limitations
+    under the License.
+
+
+=========
+Functions
+=========
+.. currentmodule:: pyspark.sql.connect
+
+Normal Functions
```
Comment (khalidmammadov) [inline python/docs/source/reference/pyspark.connect/functions.rst:24]: Perhaps “Standard Functions” sounds better?

File: python/pyspark/sql/connect/client.py [hunk 2]
```diff
@@ -203,6 +203,8 @@ def metadata(self) -> Iterable[Tuple[str, str]]:
         parameters will be converted to metadata except ones that are explicitly used
         by the channel.
 
+        .. versionadded:: 3.4.0
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/client.py:206]: `versionadded:: 3.4.0` is in the class already so it's fine

File: python/pyspark/sql/connect/client.py [hunk 3]
```diff
@@ -440,8 +450,12 @@ def register_udf(
         eval_type: int = PythonEvalType.SQL_BATCHED_UDF,
         deterministic: bool = True,
     ) -> str:
-        """Create a temporary UDF in the session catalog on the other side. We generate a
-        temporary name for it."""
+        """
+        Create a temporary UDF in the session catalog on the other side. We generate a
+        temporary name for it.
+
+        .. versionadded:: 3.4.0
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/client.py:457]: ditto let's remove them

File: python/pyspark/sql/connect/session.py [hunk 4]
```diff
@@ -106,9 +106,13 @@ def config(
         def master(self, master: str) -> "SparkSession.Builder":
             return self
 
+        master.__doc__ = PySparkSession.builder.master.__doc__
+
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/session.py:110]: let's remove these too. we should actually remove these methods away later.

File: python/pyspark/sql/connect/session.py [hunk 5]
```diff
@@ -106,9 +106,13 @@ def config(
...
         def appName(self, name: str) -> "SparkSession.Builder":
             return self.config("spark.app.name", name)
 
+        appName.__doc__ = PySparkSession.builder.appName.__doc__
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/session.py:114]: ditto

File: python/pyspark/sql/readwriter.py [hunk 6]
```diff
@@ -1934,6 +1934,9 @@ def using(self, provider: str) -> "DataFrameWriterV2":
         """
         Specifies a provider for the underlying output data source.
         Spark's default catalog supports "parquet", "json", etc.
+
+        .. versionchanged:: 3.4.0
+            Support Spark Connect.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:1939]: ditto `.. versionchanged:: 3.4.0` is already mentioned above in the class

File: python/pyspark/sql/session.py [hunk 7]
```diff
@@ -303,6 +306,9 @@ def master(self, master: str) -> "SparkSession.Builder":
 
             .. versionadded:: 2.0.0
 
+            .. versionchanged:: 3.4.0
+                Support Spark Connect.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/session.py:310]: Let's remove. Spark Connect doesnt' need master, and it doesn't work together

File: python/pyspark/sql/window.py [hunk 8]
```diff
@@ -207,6 +210,9 @@ def rowsBetween(start: int, end: int) -> "WindowSpec":
 
         .. versionadded:: 2.1.0
 
+        .. versionchanged:: 3.4.0
```
Comment (HyukjinKwon) [inline python/pyspark/sql/window.py:213]: ditto I believe this is already mentioned above in the class doc.

**Comment by itholic** [comment]
Thanks, @HyukjinKwon !

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Comment by zhengruifeng** [comment]
Late LGTM

**Pull request closed without merging**